swatch-billable-usage/src/main/java/com/redhat/swatch/billable/usage/services/BillableUsageService.java

```diff
@@ -0,0 +1,224 @@
+        log.debug("Adjusting usage based on contracted amount of {}", contractValue);
+      } catch (ContractMissingException ex) {
+        if (usage.getSnapshotDate().isAfter(OffsetDateTime.now().minus(30, ChronoUnit.MINUTES))) {
+          log.warn(
+              "{} - Unable to retrieve contract for usage less than {} minutes old. Usage: {}",
+              ErrorCode.CONTRACT_NOT_AVAILABLE,
+              30,
+              usage);
+        } else {
+          log.error(
+              "{} - Unable to retrieve contract for usage older than {} minutes old. Usage: {}",
+              ErrorCode.CONTRACT_NOT_AVAILABLE,
+              30,
+              usage);
+        }
+        return null;
+      } catch (IllegalStateException ise) {
+        log.error(
+            "Unable to retrieve contract for usage - Usage: {} Reason: {}",
+            usage,
+            ise.getMessage());
+        return null;
+      }
+    }
+
+    Quantity<BillingUnit> contractAmount =
+        Quantity.fromContractCoverage(usage, contractValue.get());
+    double applicableUsage =
+        Quantity.of(usage.getCurrentTotal())
+            .subtract(contractAmount)
+            .positiveOrZero() // ignore usage less than the contract amount
+            .getValue();
+
+    var totalRemitted = getTotalRemitted(usage);
+    BillableUsageCalculation usageCalc =
+        calculateBillableUsage(applicableUsage, usage, totalRemitted);
+
+    log.debug(
+        "Processing monthly billable usage: Usage: {}, Applicable: {}, Current total: {}, Current remittance: {}, New billable: {}",
+        usage,
+        applicableUsage,
+        usage.getCurrentTotal(),
+        totalRemitted,
+        usageCalc);
+
+    // Update the reported usage value to the newly calculated one.
+    usage.setValue(usageCalc.getBillableValue());
+    usage.setBillingFactor(usageCalc.getBillingFactor());
+
+    if (usageCalc.getRemittedValue() > 0) {
+      createRemittance(usage, usageCalc);
+    } else {
+      log.debug("Nothing to remit. Remittance record will not be created.");
+    }
+
+    // There were issues with transmitting usage to AWS since the cost event timestamps were in the
+    // past. This modification allows us to send usage to AWS if we get it during the current hour
+    // of event tally.
+    usage.setSnapshotDate(usageCalc.getRemittanceDate());
+
+    log.info(
+        "Finished producing monthly billable for orgId={} productId={} metric={} provider={}, snapshotDate={}",
+        usage.getOrgId(),
+        usage.getProductId(),
+        usage.getMetricId(),
+        usage.getBillingProvider(),
+        usage.getSnapshotDate());
+    return usage;
+  }
+
+  private double getTotalRemitted(BillableUsage billableUsage) {
+    var filter = BillableUsageRemittanceFilter.fromUsage(billableUsage);
+    return billableUsageRemittanceRepository.getRemittanceSummaries(filter).stream()
+        .findFirst()
```

Comment:
Need to be careful here because we had patched production with a hotfix that is not applied to main yet. The relevant PR for [main is here.](https://github.com/RedHatInsights/rhsm-subscriptions/pull/3301/files) Will likely be handled by git once it is put in, but we should make sure that we don't 'un-fix' the fix. :)

Reply:
Indeed :(